agent/pulp/agent/gofer/pulpplugin.py

```diff
@@ -55,12 +61,21 @@ def init_plugin():
       1. Update the plugin configuration using the consumer configuration.
       2. Register the consumer certificate bundle path for monitoring.
       3. Start the path monitor.
+      4. Validate registration.
+      5. If registered, update settings.
     """
     path = os.path.join(cfg.filesystem.id_cert_dir, cfg.filesystem.id_cert_filename)
     path_monitor.add(path, certificate_changed)
     path_monitor.start()
-    attach = Attach()
-    attach.start()
+    while True:
+        try:
+            validate_registration()
+            if registered:
+                update_settings()
+            # DONE
+            break
+        except ValidateRegistrationFailed:
```

Comment:
The validation failure is logged where it is raised.  Logging something here would be redundant.